[folly] build failure

### Operating system

Fedora Linux 38 (Container Image)

### Compiler

gcc (GCC) 13.1.1 20230511 (Red Hat 13.1.1-2)

### Steps to reproduce the behavior

```Shell
./vcpkg install folly
```


### Failure logs

```
root@NixOS-RoT ~/vcpkg (master) [1]# ./vcpkg install folly
Computing installation plan...
The following packages will be built and installed:
    folly[core,zlib]:x64-linux -> 2022.10.31.00#6
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /root/.cache/vcpkg/archives in 6.92 us. Use --debug to see more details.
Installing 1/1 folly:x64-linux...
Building folly[core,zlib]:x64-linux...
-- Using cached facebook-folly-d8ed9cd2869c74b00fa6f1a7603301183f5c2249.tar.gz.
-- Cleaning sources at /root/vcpkg/buildtrees/folly/src/183f5c2249-df85b9f5e8.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /root/vcpkg/downloads/facebook-folly-d8ed9cd2869c74b00fa6f1a7603301183f5c2249.tar.gz
-- Applying patch reorder-glog-gflags.patch
-- Applying patch disable-non-underscore-posix-names.patch
-- Applying patch boost-1.70.patch
-- Applying patch fix-windows-minmax.patch
-- Applying patch fix-deps.patch
-- Using source at /root/vcpkg/buildtrees/folly/src/183f5c2249-df85b9f5e8.clean
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j7
    Working Directory: /root/vcpkg/buildtrees/folly/x64-linux-dbg
    See logs for more information:
      /root/vcpkg/buildtrees/folly/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/folly/portfile.cmake:69 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building folly:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+folly
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[folly]+Build+error&body=Copy+issue+body+from+%2Froot%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

```

### Additional context

[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/11542122/install-x64-linux-dbg-out.log)

--- a/ports/folly/fix-build-with-gcc-13.patch
+++ b/ports/folly/fix-build-with-gcc-13.patch
@@ -1,0 +1,27 @@
+From e3cba5dd4f59c695d9cbf6bd02249af7103cc300 Mon Sep 17 00:00:00 2001
+From: Sam James <sam@gentoo.org>
+Date: Sun, 22 Jan 2023 05:06:16 +0000
+Subject: [PATCH] Fix build with GCC 13 (add missing includes)
+
+GCC 13 (as usual for new compiler releases) shuffles around some
+internal includes and so <stdexcept> etc is no longer transitively included.
+
+Signed-off-by: Sam James <sam@gentoo.org>
+---
+ folly/system/AtFork.cpp | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/folly/system/AtFork.cpp b/folly/system/AtFork.cpp
+index e888e52858a..a5570330dc3 100644
+--- a/folly/system/AtFork.cpp
++++ b/folly/system/AtFork.cpp
+@@ -14,6 +14,9 @@
+  * limitations under the License.
+  */
+ 
++#include <stdexcept>
++#include <system_error>
++
+ #include <folly/system/AtFork.h>
+ 
+ #include <folly/ScopeGuard.h>

--- a/ports/folly/portfile.cmake
+++ b/ports/folly/portfile.cmake
@@ -17,6 +17,7 @@ vcpkg_from_github(
         boost-1.70.patch
         fix-windows-minmax.patch
         fix-deps.patch
+	fix-build-with-gcc-13.patch
 )
 
 file(REMOVE "${SOURCE_PATH}/CMake/FindFmt.cmake")

--- a/ports/folly/vcpkg.json
+++ b/ports/folly/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "folly",
   "version-string": "2022.10.31.00",
-  "port-version": 6,
+  "port-version": 7,
   "description": "An open-source C++ library developed and used at Facebook. The library is UNSTABLE on Windows",
   "homepage": "https://github.com/facebook/folly",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2558,7 +2558,7 @@
     },
     "folly": {
       "baseline": "2022.10.31.00",
-      "port-version": 6
+      "port-version": 7
     },
     "font-chef": {
       "baseline": "1.1.0",

--- a/versions/f-/folly.json
+++ b/versions/f-/folly.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "003ae8b1bc9dc0a460b5c6f6cacda76fa5931cf8",
+      "version-string": "2022.10.31.00",
+      "port-version": 7
+    },
+    {
       "git-tree": "204d88dbc53dc5ff37c58459c1af0c6f19f446db",
       "version-string": "2022.10.31.00",
       "port-version": 6

